src/merge.js

```diff
@@ -55,8 +55,12 @@ function mergeJsonGraphEnvelope(
       ? mergeJsonGraph(left.jsonGraph, right.jsonGraph)
       : left.jsonGraph
   };
-  if (left.paths && right.paths) {
-    result.paths = left.paths.concat(right.paths);
+
+  const leftPaths = left.paths || [];
+  const rightPaths = right.paths || [];
+
+  if (Array.isArray(left.paths) || Array.isArray(right.paths)) {
```

Comment:
Could this result in empty concat operations? Would that cause perf issues if we do this frequently?

Reply:
Good point: https://jsperf.com/empty-concat-condition
Optimization pushed